src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,54 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 1;
+    private final int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        int start = 0;
+        int workLoad = array.length;
+
+        if (workLoad > THRESHOLD) {
```

Comment:
The variable 'workLoad' is a bit misleading as it actually represents the length of the array to be sorted. Consider renaming it to 'length' or 'arrayLength' for clarity.